cli/dashboard_device_commands.go

```diff
@@ -4,143 +4,304 @@
 package main
 
 import (
-	"net/netip"
-	"strings"
+	"context"
+	"github.com/husarnet/husarnet/cli/v2/output"
+	"github.com/husarnet/husarnet/cli/v2/requests"
+	"github.com/husarnet/husarnet/cli/v2/types"
+	"slices"
 
-	"github.com/husarnet/husarnet/cli/v2/generated"
-
-	"github.com/Khan/genqlient/graphql"
-	u "github.com/rjNemo/underscore"
-	"github.com/urfave/cli/v2"
+	"github.com/urfave/cli/v3"
 )
 
-func interactiveGetDeviceIpByName() string {
-	response := callDashboardAPI(func(client graphql.Client) (*generated.GetDevicesResponse, error) {
-		return generated.GetDevices(client)
-	})
-
-	deviceNames := u.Map(response.Devices, func(d generated.GetDevicesDevicesDeviceType) string { return d.Name })
+var dashboardDeviceListCommand = &cli.Command{
+	Name:      "list",
+	Aliases:   []string{"ls"},
+	Usage:     "Display a table of all your devices",
+	ArgsUsage: " ", // No arguments needed
+	Action: func(ctx context.Context, cmd *cli.Command) error {
+		ignoreExtraArguments(cmd)
 
-	selectedDevice := interactiveChooseFrom("Select device", deviceNames)
+		resp, err := requests.FetchDevices()
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
 
-	device, err := u.Find(response.Devices, func(d generated.GetDevicesDevicesDeviceType) bool { return d.Name == selectedDevice })
-	if err != nil {
-		dieE(err)
-	}
+		if rawJson {
+			output.PrintJsonOrError(resp)
+			return nil
+		}
 
-	return device.DeviceId
+		prettyPrintDevices(resp)
+		return nil
+	},
 }
 
-func getDeviceIpByName(name string) string {
-	response := callDashboardAPI(func(client graphql.Client) (*generated.GetDevicesResponse, error) {
-		return generated.GetDevices(client)
-	})
+var dashboardDeviceShowCommand = &cli.Command{
+	Name:      "show",
+	Usage:     "Display device details",
+	ArgsUsage: "<device uuid OR Husarnet IPv6 addr OR hostname>",
+	Action: func(ctx context.Context, cmd *cli.Command) error {
+		requiredArgumentsNumber(cmd, 1)
+		arg := cmd.Args().First()
+		uuid, err := determineDeviceUuid(arg)
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
 
-	device, err := u.Find(response.Devices, func(d generated.GetDevicesDevicesDeviceType) bool { return d.Name == name })
-	if err != nil {
-		dieE(err)
-	}
+		resp, err := requests.FetchDeviceByUuid(uuid)
+		if err != nil {
+			printError(err.Error())
+			return nil
+		}
 
-	return device.DeviceId
+		if rawJson {
+			output.PrintJsonOrError(resp)
+		} else {
+			prettyPrintDevice(resp)
+		}
+		return nil
+	},
 }
 
-func getDeviceIpByNameOrIp(candidate string) string {
-	addr, err := netip.ParseAddr(candidate)
-	if err == nil {
-		return addr.String()
-	}
-	return getDeviceIpByName(candidate)
-}
+var dashboardDeviceUpdateCommand = &cli.Command{
+	Name:      "update",
+	Usage:     "Update device details",
```

Comment:
Would be nice to mention that **this** device will be updated in case no device specified